pkg/registry/core/node/strategy.go

```diff
@@ -127,14 +127,23 @@ type nodeStatusStrategy struct {
 var StatusStrategy = nodeStatusStrategy{Strategy}
 
 func (nodeStatusStrategy) PrepareForCreate(ctx context.Context, obj runtime.Object) {
-	_ = obj.(*api.Node)
+	node := obj.(*api.Node)
 	// Nodes allow *all* fields, including status, to be set on create.
+
+	if !utilfeature.DefaultFeatureGate.Enabled(features.DynamicKubeletConfig) {
```

Comment:
since I am not familiar with the pgk/registry code-base, why is this required?  Same question for addition in PrepareForUpdate.

Reply:
We strip alpha fields when the corresponding feature gate is turned off, so that they can't be written unless the feature gate is turned on. See similar code in `func (nodeStrategy) PrepareForCreate/PrepareForUpdate`, similarly see `DropDisabledAlphaFields` in `pkg/api/pod/util.go`.

Reply:
great, thanks